tests/andromede/test_quota_co2.py

```diff
@@ -0,0 +1,194 @@
+        ModelPort(port_type=FLOW_PORT, port_name="port_to"),
+    ],
+    port_fields_definitions=[
+        PortFieldDefinition(
+            port_field=PortFieldId("port_from", "F"),
+            definition=-var("flow"),
+        ),
+        PortFieldDefinition(
+            port_field=PortFieldId("port_to", "F"),
+            definition=var("flow"),
+        )
+    ]
+)
+
+
+"""
+build the quota CO² test system.
+
+    N1 -----N2----Demand         ^
+    |       |
+    Oil1    Coal1
+    |       |
+    ---------
+        |
+    MonQuotaCO2
+
+"""
+def test_quota_co2():
+    n1 = Node(model=NODE_MODEL, id="N1")
+    n2 = Node(model=NODE_MODEL, id="N2")
+    oil1 = create_component(model=C02_POWER_MODEL, id="Oil1")
+    coal1 = create_component(model=C02_POWER_MODEL, id="Coal1")
+    l12 = create_component(model=LINK_MODEL, id='L12')
+    demand = create_component(model=DEMAND_MODEL, id='Demand')
+    monQuotaCO2 = create_component(model=QUOTA_CO2_MODEL, id='MonQuotaCO2')
+
+    network = Network("test")
+    network.add_node(n1)
+    network.add_node(n2)
+    network.add_component(oil1)
+    network.add_component(coal1)
+    network.add_component(l12)
+    network.add_component(demand)
+    network.add_component(monQuotaCO2)
+
+    network.connect(PortRef(demand, "FlowD"), PortRef(n2, "FlowN"))
+    network.connect(PortRef(n2, "FlowN"), PortRef(l12, "port_from"))
+    network.connect(PortRef(l12, "port_to"), PortRef(n1, "FlowN"))
+    network.connect(PortRef(n1, "FlowN"), PortRef(oil1, "FlowP"))
+    network.connect(PortRef(n2, 'FlowN'), PortRef(coal1, "FlowP"))
+    network.connect(PortRef(oil1, "OutCO2"), PortRef(monQuotaCO2, "emissionCO2"))
+    network.connect(PortRef(coal1, "OutCO2"), PortRef(monQuotaCO2, "emissionCO2"))
+
+    database = DataBase()
+    database.add_data("Demand", "d", ConstantData(100))
+    database.add_data("Coal1", "p_min", ConstantData(0))
+    database.add_data("Oil1", "p_min", ConstantData(0))
+    database.add_data("Coal1", "p_max", ConstantData(100))
+    database.add_data("Oil1", "p_max", ConstantData(100))
+    database.add_data("Coal1", "taux_emission", ConstantData(2))
+    database.add_data("Oil1", "taux_emission", ConstantData(1))
+    database.add_data("Coal1", "cost", ConstantData(10))
+    database.add_data("Oil1", "cost", ConstantData(100))
+    database.add_data("L12", "f_max", ConstantData(100))
+    # when the bug in the port is fixed, the quota should be 150
+    database.add_data("MonQuotaCO2", "quota", ConstantData(150))
+
+    scenarios = 1
+    problem = build_problem(network, database, TimeBlock(1, [0]), scenarios)
+    status = problem.solver.Solve()
+
+    output = OutputValues(problem)
+    oil1_p = output.component('Oil1').var('p').value
+    coal1_p = output.component('Coal1').var('p').value
+    l12_flow = output.component('L12').var('flow').value
+
+    assert status == problem.solver.OPTIMAL
+    assert problem.solver.Objective().Value() == 5500
```

Comment:
Warning when comparing floats for equality, you should rather use `is_close` from `OutputValues`